src/main/scala/com/fulcrumgenomics/bam/api/SamRecord.scala

```diff
@@ -184,10 +184,12 @@ trait SamRecord {
   @inline final def readGroup: SAMReadGroupRecord = getReadGroup
 
   // Use apply/update for tag attributes
-  @inline final def apply[A](name: String): A              = getAttribute(name).asInstanceOf[A]
-  @inline final def get[A](name: String): Option[A]        = Option(apply(name))
-  @inline final def update(name: String, value: Any): Unit = setAttribute(name, value)
-  @inline final def attributes: Map[String,Any] = getAttributes.map(x => x.tag -> x.value).toMap
+  @inline final def apply[A](name: String): A                 = getAttribute(name).asInstanceOf[A]
+  @inline final def get[A](name: String): Option[A]           = Option(apply(name))
+  @inline final def update(name: String, value: Any): Unit    = setAttribute(name, value)
+  @inline final def attributes: Map[String,Any]               = getAttributes.map(x => x.tag -> x.value).toMap
+  @inline final def getOrElse[A](name: String, default: A): A = get(name).getOrElse(default)
```

Comment:
do we want default to have type `=> A` so that it mimics `Option` and also allows you to do things like `rec.getOrElse("RG", throw new IllegalStateException("Booo!"))`?